nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service/src/main/java/org/apache/nifi/ssl/StandardSSLContextService.java

```diff
@@ -185,14 +191,24 @@ public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, S
             }
         }
 
-        results.addAll(validateStore(validationContext.getProperties(), KeystoreValidationGroup.KEYSTORE));
-        results.addAll(validateStore(validationContext.getProperties(), KeystoreValidationGroup.TRUSTSTORE));
+        final Map<PropertyDescriptor, String> properties = evaluateProperties(validationContext.getProperties(), validationContext);
+        results.addAll(validateStore(properties, KeystoreValidationGroup.KEYSTORE));
+        results.addAll(validateStore(properties, KeystoreValidationGroup.TRUSTSTORE));
 
         isValidated = results.isEmpty();
 
         return results;
     }
 
+    private Map<PropertyDescriptor, String> evaluateProperties(final Map<PropertyDescriptor, String> properties, final PropertyContext context) {
```

Comment:
Is there a reason for passing the Map of properties as a separate parameter from the PropertyContext as opposed to just passing the PropertyContext and then calling getProperties() in this method?

Reply:
The `PropertyContext#getProperties` uses the property names as keys and not `PropertyDescriptor`s whereas I needed access to the latter to check whether they supported expression language

When I look at @markobean suggestion (changing the if from an expression language check to one based on explicit property names), I'll consider whether the `PropertyContext` could then be used

Reply:
Sounds good, I was mainly asking whether the method signature could be changed to just `evaluateProperties(PropertyContext context)` based on how the method was being called.

Reply:
API simplified

Reply:
Chris- you mention "API simplified" implying you updated the signature to be `evaluateProperties(PropertyContext context)`, but I don't see that update. Have you pushed your latest? Thx.

Reply:
@markobean yes, see https://github.com/apache/nifi/pull/5147/commits/a27fc7955076749e3db49c80deef0f0b7d038377#diff-d1b1db3b2c0d67997b262529940d794f98ac6a485c6a21316f7af1b935d95ffaR204 (commit 2)